src/main/groovy/5configureEmailExtension.groovy

```diff
@@ -0,0 +1,154 @@
+    "ext_mailer_max_attachment_size": emailConfig.MAX_ATTACHMENT_SIZE,
+    "ext_mailer_default_recipients": emailConfig.DEFAULT_RECIPIENTS,
+    "ext_mailer_add_precedence_bulk": emailConfig.ADD_PRECEDENCE_BULK,
+    "ext_mailer_allowed_domains": emailConfig.ALLOWED_DOMAINS,
+    "ext_mailer_excluded_committers": emailConfig.EXCLUDED_COMMITTERS,
+    "ext_mailer_require_admin_for_template_testing": emailConfig.REQUIRE_ADMIN_FOR_TEMPLATE_TESTING,
+    "ext_mailer_watching_enabled": emailConfig.WATCHING_ENABLED,
+    "ext_mailer_allow_unregistered_enabled": emailConfig.ALLOW_UNREGISTERED_ENABLED
+]
+if (['text/html', 'text/plain'].contains(emailConfig.DEFAULT_CONTENT_TYPE)) {
+    params["ext_mailer_default_content_type"] = emailConfig.DEFAULT_CONTENT_TYPE
+} else {
+    logger.severe("Invalid content type specified for DEFAULT CONTENT TYPE")
+    jenkins.doSafeExit(null)
+    System.exit(1)
+}
+if (emailConfig.USE_LIST_ID) {
+    params["ext_mailer_use_list_id"] =  emailConfig.USE_LIST_ID
+    params["ext_mailer_list_id"] =  emailConfig.LIST_ID
+}
+
+// The following values can be quite complex, and are stored in their own separate
+// files
+paths = [
+    "ext_mailer_default_presend_script": emailConfig.DEFAULT_PRESEND_SCRIPT_PATH,
+    "ext_mailer_default_postsend_script": emailConfig.DEFAULT_POSTSEND_SCRIPT_PATH,
+    "ext_mailer_default_body": emailConfig.DEFAULT_BODY_PATH
+]
+paths.each { param_name, filePath ->
+    if (!filePath || filePath == '') {
+        params[param_name] = null
+    } else {
+        try {
+            fullFilePath = "${configPath}/${filePath}"
+            params[param_name] = new File(fullFilePath).text
+        } catch (FileNotFoundException e) {
+            logger.severe("Cannot find script file path @ ${fullFilePath}")
+            jenkins.doSafeExit(null)
+            System.exit(1)
+        }
+    }
+}
+
+def stapler = [
+    getParameter: { name -> params[name] },
+    hasParameter: { name -> params.keySet().contains(name) },
+    getParameterValues: { name -> params[name] } // should be String[]
+] as org.kohsuke.stapler.StaplerRequest
+
+JSONObject json = new JSONObject()
+def triggerList = [
+    'AbortedTrigger',
+    'AbstractScriptTrigger',
+    'AlwaysTrigger',
+    'BuildingTrigger',
+    'FailureTrigger',
+    'FirstFailureTrigger',
+    'FirstUnstableTrigger',
+    'FixedTrigger',
+    'FixedUnhealthyTrigger',
+    'ImprovementTrigger',
+    'NotBuiltTrigger',
+    'NthFailureTrigger',
+    'PreBuildScriptTrigger',
+    'PreBuildTrigger',
+    'RegressionTrigger',
+    'ScriptTrigger',
+    'SecondFailureTrigger',
+    'StatusChangedTrigger',
+    'StillFailingTrigger',
+    'StillUnstableTrigger',
+    'SuccessTrigger',
+    'UnstableTrigger',
+    'XNthFailureTrigger'
+]
+// verify that only valid trigger classes are used to set default triggers
+emailConfig.TRIGGERS.each { trigger ->
+    if (! triggerList.contains(trigger)) {
+        logger.severe('NOPE')
```

Comment:
Maybe a slightly more verbose error message would be helpful here? 😃

Reply:
Ha. Oops, sorry forgot to make it nice for other people